client/clientBean.go

```diff
@@ -280,23 +248,23 @@ func (d *dnsUpdater) Start() {
 	go func() {
 		for {
 			now := time.Now()
-			newPeers, updates, changed, err := d.refresh()
+			res, err := d.refresh()
 			if err != nil {
 				d.logger.Error("Failed to update DNS", tag.Error(err), tag.Address(d.dnsAddress))
 			}
-			if changed {
-				if len(updates.Additions) > 0 {
-					d.logger.Info("Add new peers by DNS lookup", tag.Address(d.dnsAddress), tag.Addresses(identifiersToStringList(updates.Additions)))
+			if res.changed {
+				if len(res.updates.Additions) > 0 {
```

Comment:
Seems like you could just log this in the refresh function and do not need to return it..

Reply:
addressed offline